### PR TITLE
feat(telephony.alias): voip alias conference new page

### PR DESCRIPTION
**Pull request by JeremyDec**

### CODE REVIEW

File: client/telecomUniverseComponents/telecom/voip/service/alias/voip-service-alias.service.js [hunk 1]
```diff
@@ -125,8 +127,31 @@ export default class {
...
+  fetchNumberSounds({ billingAccount, serviceName }) {
+    return this.OvhApiTelephony.EasyHunting().Sound().v6().query({
+      billingAccount,
+      serviceName,
+    }).$promise.then(soundIds => this.$q.all(
+      soundIds.map(soundId => this.OvhApiTelephony.EasyHunting().Sound().v6().get({
+        billingAccount,
+        serviceName,
+        soundId,
+      })),
```
Comment (antleblanc) [inline client/telecomUniverseComponents/telecom/voip/service/alias/voip-service-alias.service.js]: I bet it was more something like this:
```suggestion
      }).$promise),
```
And `.$promise` L148 could be deleted.

File: client/app/telecom/telephony/alias/configuration/feature/conference/telecom-telephony-alias-configuration-conference.controller.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+    this.serviceInfos = {
+      billingAccount: this.$stateParams.billingAccount,
+      serviceName: this.$stateParams.serviceName,
+    };
+
+    this.loading = true;
+    return this.tucVoipServiceAlias.fetchConferenceNumber(this.serviceInfos)
+      .then((conference) => {
+        this.conference = _.omit(conference, ['$promise', '$resolved']);
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/feature/conference/telecom-telephony-alias-configuration-conference.controller.js]: Why not use `angular.copy` instead?

File: client/app/telecom/telephony/alias/configuration/feature/conference/telecom-telephony-alias-configuration-conference.controller.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        );
+        this.reportStatus = this.availableReportStatus.find(
+          ({ value }) => value === conference.reportStatus,
+        );
+
+        return this.tucVoipServiceAlias.fetchConferenceNumberWebAccess(this.serviceInfos);
+      })
+      .then((webAccess) => {
+        this.webAccess = this.formatWebAccess(webAccess);
+        this.hasWebAccessUrls = !_(webAccess).isEmpty();
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/feature/conference/telecom-telephony-alias-configuration-conference.controller.js]: Useless chaining.

File: client/app/telecom/telephony/alias/configuration/feature/conference/telecom-telephony-alias-configuration-conference.controller.js [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+        this.TUC_TELEPHONY_ALIAS.conference.webAccessType.followUp,
+      ),
+      control: this.tucVoipServiceAlias.createConferenceNumberWebAccess(
+        this.serviceInfos,
+        this.TUC_TELEPHONY_ALIAS.conference.webAccessType.control,
+      ),
+    }).then(() => this.tucVoipServiceAlias.fetchConferenceNumberWebAccess(this.serviceInfos))
+      .then((webAccess) => {
+        this.webAccess = this.formatWebAccess(webAccess);
+        this.hasWebAccessUrls = !_(webAccess).isEmpty();
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/feature/conference/telecom-telephony-alias-configuration-conference.controller.js]: Useless chaining.

File: client/app/telecom/telephony/alias/configuration/feature/conference/telecom-telephony-alias-configuration-conference.controller.js [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+    return {
+      followUp: webAccess
+        .find(({ type }) => type === this.TUC_TELEPHONY_ALIAS.conference.webAccessType.followUp),
+      control: webAccess
+        .find(({ type }) => type === this.TUC_TELEPHONY_ALIAS.conference.webAccessType.control),
+    };
+  }
+
+  canUpdateConference() {
+    const hasChanged = !_(this.conference).isEqual(this.conferenceCopy);
```
Comment (antleblanc) [inline client/app/telecom/telephony/alias/configuration/feature/conference/telecom-telephony-alias-configuration-conference.controller.js]: Useless chaining (other option: `angular.equals`).

File: client/app/telecom/telephony/alias/configuration/feature/conference/telecom-telephony-alias-configuration-conference.controller.js [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+        .find(({ type }) => type === this.TUC_TELEPHONY_ALIAS.conference.webAccessType.followUp),
+      control: webAccess
+        .find(({ type }) => type === this.TUC_TELEPHONY_ALIAS.conference.webAccessType.control),
+    };
+  }
+
+  canUpdateConference() {
+    const hasChanged = !angular.equals(this.conference, this.conferenceCopy);
+    return hasChanged || this.newFile;
+    // const hasChanged = !_.isEqual(this.conference, this.conferenceCopy);
```
Comment (marie-j) [inline client/app/telecom/telephony/alias/configuration/feature/conference/telecom-telephony-alias-configuration-conference.controller.js]: Unecessary comment

File: client/app/telecom/telephony/alias/configuration/feature/conference/telecom-telephony-alias-configuration-conference.html [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+                    </oui-popover-content>
+                </oui-popover>
+            </dt>
+            <dd data-ng-if="!$ctrl.newFile"
+                data-ng-bind="$ctrl.conference.announceFile ? $ctrl.conference.announceFilename : ('common_none' | translate)"></dd>
+            <dd>
+                <tuc-input-file data-ng-model="$ctrl.newFile"
+                                data-ng-accept="audio/*"
+                                data-ng-accept-filter="$ctrl.checkValidAudioExtention(file)">
+                    <span class="fa fa-upload mr-1"></span>
```
Comment (marie-j) [inline client/app/telecom/telephony/alias/configuration/feature/conference/telecom-telephony-alias-configuration-conference.html]: I suppose there is no `oui-icon` equivalent ?

Comment (JeremyDec) [reply client/app/telecom/telephony/alias/configuration/feature/conference/telecom-telephony-alias-configuration-conference.html]: No there is not

File: client/app/telecom/telephony/alias/configuration/feature/conference/telecom-telephony-alias-configuration-conference.html [hunk 8]
```diff
@@ -125,44 +125,50 @@
...
-
-        <div data-ng-if="!$ctrl.generatingUrls">
-            <button class="oui-button oui-button_secondary oui-button_icon-left mb-5"
-                    data-ng-if="!$ctrl.hasWebAccessUrls"
-                    data-ng-click="$ctrl.generateWebAccess()">
-                <span class="oui-icon oui-icon-add"
-                      aria-hidden="true"></span>
-                <span data-translate="telephony_alias_config_conference_generate_urls"></span>
-            </button>
+    <label class="oui-label"
```
Comment (FredericEspiau) [inline client/app/telecom/telephony/alias/configuration/feature/conference/telecom-telephony-alias-configuration-conference.html]: `label` linked to nothing (where is the `for`)

File: client/telecomUniverseComponents/telecom/voip/service/alias/voip-service-alias.service.js [hunk 9]
```diff
@@ -161,16 +186,152 @@ export default class {
...
+          serviceName,
+          taskId,
+          {
+            namespace: `uploadConferenceNumberAnnounceTask_${serviceName}`,
+            interval: 1000,
+            retryMaxAttempts: 0,
+          },
+        ))
+      .then(() => true)
+      .catch(error => error);
```
Comment (FredericEspiau) [inline client/telecomUniverseComponents/telecom/voip/service/alias/voip-service-alias.service.js]: Is this intentional ? Handling errors in `then` is not a good idea, except maybe if the errors returned by the API are actually not errors

File: client/telecomUniverseComponents/telecom/voip/service/alias/voip-service-alias.service.js [hunk 10]
```diff
@@ -5,25 +5,27 @@ import _ from 'lodash';
...
+    this.OvhApiMe = OvhApiMe;
     this.OvhApiTelephony = OvhApiTelephony;
     this.voipServiceTask = voipServiceTask;
   }
 
   /**
    *  @ngdoc method
-   *  @name managerApp.service:tucVoipServiceAlias#getConvertToLineTask
-   *  @methodOf managerApp.service:tucVoipServiceAlias
+   *  @name managerApp.service:voipServiceAlias#getConvertToLineTask
```
Comment (FredericEspiau) [inline client/telecomUniverseComponents/telecom/voip/service/alias/voip-service-alias.service.js]: Intentional ?

File: client/app/telecom/telephony/alias/configuration/translations/Messages_fr_FR.xml [hunk 11]
```diff
@@ -109,4 +109,43 @@
    <translation id="telephony_alias_config_sip_line_activate_presentation">Activer la présentation de votre numéro lors des appels sortants</translation>
    <translation id="telephony_alias_config_sip_line_activate_presentation_warning">Attention : si vous avez paramétré la présentation d’un autre numéro sur la ligne, il sera remplacé par ce numéro.</translation>
 
+   <!-- Conférence -->
```
Comment (FredericEspiau) [inline client/app/telecom/telephony/alias/configuration/translations/Messages_fr_FR.xml]: Content ? 😄

Comment (FredericEspiau) [reply client/app/telecom/telephony/alias/configuration/translations/Messages_fr_FR.xml]: also, `Conférence` is French

File: client/app/telecom/telephony/alias/configuration/feature/conference/telecom-telephony-alias-configuration-conference.controller.js (hunk 2 above)
Comment (FredericEspiau) [reply client/app/telecom/telephony/alias/configuration/feature/conference/telecom-telephony-alias-configuration-conference.controller.js]:
```suggestion
        { $promis, $resolved, ...this.conference } = conference;
```

Comment (euhmeuh) [reply client/app/telecom/telephony/alias/configuration/feature/conference/telecom-telephony-alias-configuration-conference.controller.js]: IMOH I find `_.omit` easier to understand. I know we're trying to get rid of underscore in the long run but, it just conveys meaning better than declaring some variables and not using them, I guess.

File: client/app/telecom/telephony/alias/configuration/feature/conference/telecom-telephony-alias-configuration-conference.html (hunk 8 above)
Comment (JeremyDec) [reply client/app/telecom/telephony/alias/configuration/feature/conference/telecom-telephony-alias-configuration-conference.html]: No element to link to => going to be a `<span>`

File: client/telecomUniverseComponents/telecom/voip/service/alias/voip-service-alias.service.js (hunk 10 above)
Comment (JeremyDec) [reply client/telecomUniverseComponents/telecom/voip/service/alias/voip-service-alias.service.js]: No at all :)

File: client/app/telecom/telephony/alias/configuration/translations/Messages_fr_FR.xml (hunk 11 above)
Comment (JeremyDec) [reply client/app/telecom/telephony/alias/configuration/translations/Messages_fr_FR.xml]: good catch

File: client/app/telecom/telephony/alias/configuration/feature/conference/telecom-telephony-alias-configuration-conference.html (hunk 7 above)
Comment (FredericEspiau) [reply client/app/telecom/telephony/alias/configuration/feature/conference/telecom-telephony-alias-configuration-conference.html]:
```suggestion
                    <span class="fa fa-upload mr-1" aria-hidden="true"></span>
```

**Pull request merged**